src/problems/basic_problems.jl

```diff
@@ -411,52 +410,47 @@ OptimizationProblem{iip}(f, u0, p = SciMLBase.NullParameters(),;
-if you set a `callback` in the problem, then that `callback` will be added in
-every solve call.
+parameters.
 
 `lb` and `ub` are the upper and lower bounds for box constraints on the
-optimization. They should be an `AbstractArray` matching the geometry of `u`,
-where `(lb[I],ub[I])` is the box constraint (lower and upper bounds)
-for `u[I]`.
-
-`lcons` and `ucons` are the upper and lower bounds for equality constraints on the
-optimization. They should be an `AbstractArray` matching the geometry of `u`,
-where `(lcons[I],ucons[I])` is the constraint (lower and upper bounds)
-for `cons[I]`.
+optimization variables. They should be an `AbstractArray` matching the geometry of `u`,
+where `(lb[i],ub[i])` is the box constraint (lower and upper bounds) for `u[i]`.
 
-If `f` is a standard Julia function, it is automatically transformed into an 
-`OptimizationFunction` with `NoAD()`, meaning the derivative functions are not 
-automatically generated.
+`lcons` and `ucons` are the upper and lower bounds in case of inequality constraints on the
+optimization and if they are set to be equal then it represents an equality constraint.
+They should be an `AbstractArray` matching the geometry of `u`, where `(lcons[i],ucons[i])`
+are the lower and upper bounds for `cons[i]`.
 
-Any extra keyword arguments are captured to be sent to the optimizers.
+The `f` in the `OptimizationProblem` should typically be an instance of [`OptimizationFunction`](@ref)
+to specify the objective function and its derivatives either by passing predefined functions for them or
+automatically generated using the [`ADType`](@ref). If `f` is a standard Julia function,
+it is passed on as is to the optimization solver.
 
 ### Fields
 
 * `f`: the function in the problem.
-* `u0`: the initial guess for the optima.
-* `p`: the parameters for the problem. Defaults to `NullParameters`.
-* `lb`: the lower bounds for the optimization of `u`.
-* `ub`: the upper bounds for the optimization of `u`.
-* `int`: integrality indicator for `u`.
+* `u0`: the initial guess for the optimization variables.
+* `p`: the constant parameters used for defining the problem. Defaults to `NullParameters`.
+* `lb`: the lower bounds for the optimization variables `u`.
+* `ub`: the upper bounds for the optimization variables `u`.
+* `int`: integrality indicator for `u`. If `int[i] == 1`, then `u[i]` is an integer variable.
+    Defaults to `nothing`, implying no integrality constraints.
 * `lcons`: the vector of lower bounds for the constraints passed to [OptimizationFunction](@ref).
     Defaults to `nothing`, implying no lower bounds for the constraints (i.e. the constraint bound is `-Inf`)
 * `ucons`: the vector of upper bounds for the constraints passed to [OptimizationFunction](@ref).
     Defaults to `nothing`, implying no upper bounds for the constraints (i.e. the constraint bound is `Inf`)
 * `sense`: the objective sense, can take `MaxSense` or `MinSense` from Optimization.jl.
```

Comment:
```suggestion
* `sense`: the objective sense, can take `MaxSense` or `MinSense` from Optimization.jl.
* `kwargs`: the keyword arguments passed on to the solvers.
```

Reply:
Nope, the kwargs passed on to the solvers are the ones to `solve`

Reply:
and the prob.kwargs... or Optimization.jl is wrong.

Reply:
Fixed https://github.com/SciML/SciMLBase.jl/pull/432